forum/communications/consumers.py

```diff
@@ -0,0 +1,48 @@
+from channels.generic.websocket import AsyncJsonWebsocketConsumer
+from users.models import CustomUser
+
+
+class ChatConsumer(AsyncJsonWebsocketConsumer):
+    async def connect(self):
+        self.room_name = self.scope["url_route"]["kwargs"]["room_name"]
+        self.room_group_name = f"chat_{self.room_name}"
+
+        await self.channel_layer.group_add(
+            self.room_group_name, self.channel_name
+        )
+
+        await self.accept()
+
+    async def disconnect(self, code: int):
+        await self.channel_layer.group_discard(
+            self.room_group_name, self.channel_name
+        )
```

Comment:
Ensure proper cleanup of resources or states if needed, beyond just removing the channel from the group